### ISSUE TITLE
Allow "stdin mode"

### ISSUE THREAD

**Issue by fancsali** [nils-werner/crestic#18]
One of the very nice, and convenient features of `restic` is, that one can do a backup for the output of a command, like:

```
mysql-dump --all-databases | restic backup --stdin --stdin-filename mysql-all.sql
```

(See https://restic.readthedocs.io/en/latest/040_backup.html#reading-data-from-stdin)

It would be brilliant, if `crestic` would be able to build such commands as well.

**Comment by nils-werner** [comment]
What have you tried so far? Given the correct config file (no arguments that conflict with `--stdin`) I believe it should just work...

**Comment by fancsali** [comment]
Indeed, it doesn't. What I meant, is that it might be a bit error-prone, to configure things in two places:

Once in `/etc/crestic.cfg`:

```
[mysql]
repo: ...
keep-daily: ...
...
```

And then a custom systemd unit for this.

So, I had the idea, to provide a method of inserting something at the very beginning of the assembled command line.

**Comment by nils-werner** [comment]
Hm, can you please give a reproducible example and the error message you are seeing?

**Comment by nils-werner** [comment]
> So, I had the idea, to provide a method of inserting something at the very beginning of the assembled command line.

Or did you mean you want the `mysql-dump --all-databases | ` part created by `crestic`?

**Comment by fancsali** [comment]
> > So, I had the idea, to provide a method of inserting something at the very beginning of the assembled command line.
> 
> Or did you mean you want the `mysql-dump --all-databases | ` part created by `crestic`?

Well, "want" is a very strong word here, but yes, I am proposing that might be a nice feature... 😉

**Comment by nils-werner** [comment]
Ah, I see. That is a shell feature and therefore probably out of scope for `crestic`, when all you need to do is

    mysql-dump --all-databases | crestic mysql

**Comment by fancsali** [comment]
Well, it might be out of scope; it just occured to me.

A solution around it is to make 2 systemd templates, one for when there's a command to pipe, and one for "standard" invocation.
